### DISCUSSION TITLE
Bevy Retro Integration Progress

### DISCUSSION THREAD

**Discussion by zicklag** [RAUI-labs/raui#17] [Show and tell]
I've made great progress integrating RAUI with [Bevy Retro](https://github.com/katharostech/bevy_retro) and I'll be sharing my progress and thoughts here as I go.

**Comment by zicklag** [comment]
So far I'm making use of the tesselator renderer, which is awesome BTW, and I think I've gotten the UI rendering completely how I need it. There might be some rough edges in the renderer that I need fix, but otherwise it's working! Here's a quick demo with a simple horizontal layout and a colored square, a text box, and an image:

![bevyRetroUi](https://user-images.githubusercontent.com/25393315/116586693-5b6d5f80-a8df-11eb-818f-16cbd6b5ec72.gif)

Now I have to get better at using RAUI so I can make a better demo, similar to the `in-game` example RAUI already has. I also have to the interaction engine working, which should be simple.

So far, integration is surprisingly easy, good job @PsichiX !

**Reply by PsichiX** [reply]
i'm glad to hear that, one of the most important RAUI goals was to make it super easy to integrate into any renderer as possible, i'm happy it achieves this goal :D

**Comment by zicklag** [comment]
Got framed images and RAUI material working 🎉

![bevy_retro_ui](https://user-images.githubusercontent.com/25393315/116764891-3e708380-a9e8-11eb-8ca6-8e9e1333a3e9.gif)

**Reply by PsichiX** [reply]
perfect <3

**Comment by zicklag** [comment]
Interactions are now implemented for Bevy Retro! ( Pixelation is because of Bevy Retro's resolution )

![Peek 2021-05-05 15-29](https://user-images.githubusercontent.com/25393315/117205145-d5598900-adb6-11eb-85ad-f2cc018d7cea.gif)

**Reply by PsichiX** [reply]
oh my, that's awesome! :O :D

**Reply by zicklag** [reply]
Now I've just got to get better at using it. :)

Thanks for the toolkit @PsichiX !

**Comment by zicklag** [comment]
Finally figured out how to get the buttons clickable in my own example. I had missed the navigation enabled containers. :D

![working-button](https://user-images.githubusercontent.com/25393315/118317794-1c820100-b4be-11eb-9183-96ae390cab56.gif)

**Comment by zicklag** [comment]
Successfully integrated with ECS using the `ProcessContext` feature added in #62. I have an example now with the button sending events to the Bevy ECS, and I also have another example where I grab FPS and sprite counter resources from the Bevy ECS in a benchmark!

In this example the button click sends events to the ECS which uses an ECS system to move the backgound map when the button is clicked.

![ui](https://user-images.githubusercontent.com/25393315/118345414-ef534400-b4f9-11eb-81bc-cdc2287f495b.gif)

**Comment by zicklag** [comment]
Using RAUI in our actual game! Working great so far! :tada:


![raui-in-bevy-retro](https://user-images.githubusercontent.com/25393315/119063587-18af1c80-b99f-11eb-8f10-fa575fb712fb.gif)

**Reply by PsichiX** [reply]
it's so exciting to see :D
